### PR TITLE
View stats view

### PR DESCRIPTION
**Pull request by meschac38700**

### CODE REVIEW

File: pod/video/templates/videos/video_stats_view.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{% load staticfiles i18n %}
+{% block opengraph %}
+    {{ block.super }}
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+{% endblock opengraph %}
+{% block page_title %}{% trans "Video - Statistics views" %}{% endblock page_title %}
+{% block page_extra_head %}
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.11.4/themes/redmond/jquery-ui.min.css">
```
Comment (ptitloup) [inline pod/video/templates/videos/video_stats_view.html]: Call it in local

File: pod/video/templates/videos/video_stats_view.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/free-jqgrid/4.15.5/css/ui.jqgrid.min.css">
```
Comment (ptitloup) [inline pod/video/templates/videos/video_stats_view.html]: Call it in local server

File: pod/video/templates/videos/video_stats_view.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/free-jqgrid/4.15.5/jquery.jqgrid.min.js"></script>
```
Comment (ptitloup) [inline pod/video/templates/videos/video_stats_view.html]: Call it in local server

File: pod/video/templates/videos/video_stats_view.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    <!-- https://free-jqgrid.github.io/getting-started/index.html -->
```
Comment (ptitloup) [inline pod/video/templates/videos/video_stats_view.html]: remove it

File: pod/video/templates/videos/video_stats_view.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+			})
+			// remove date_min in data
+			data.pop();
+			document.querySelector("#jsperiode").min = min_date[0].min_date;
+		},
+		postData: {
+			csrfmiddlewaretoken: $("[name=csrfmiddlewaretoken]").val()
+		}
+        });
+	let today = new Date().toISOString().split("T")[0];
```
Comment (ptitloup) [inline pod/video/templates/videos/video_stats_view.html]: Please test it in multiple browser (google, firefox, safari and IE) and tell me if it's ok

Comment (ptitloup) [reply pod/video/templates/videos/video_stats_view.html]: I'm not sure that "let" declaration is ok in all browser

File: pod/video/views.py [hunk 6]
```diff
@@ -35,12 +38,13 @@
...
 import re
 import pandas
-from datetime import date
+from datetime import date, datetime
 
 from pod.playlist.models import Playlist
 from django.db import transaction
 from django.db import IntegrityError
 
+TODAY = datetime.now()
```
Comment (ptitloup) [inline pod/video/views.py]: is it ok (not in cache ?)

Comment (ptitloup) [reply pod/video/views.py]: and why not use date.today() ?

File: pod/video/views.py [hunk 7]
```diff
@@ -1282,3 +1286,111 @@ def video_oembed(request):
...
+# Retourne une ou plusieurs videos et le titre de
+# (theme, ou video ou channel ou videos pour toutes)
+# selon la réference du slug donnée
+# (video ou channel ou theme ou videos pour toutes les videos)
+def get_videos(p_slug, target, p_slug_t=None):
+    videos = []
+    title = _("Pod video view statistics")
+    if target.lower() == "video":
+        videos.append(
+                Video.objects.filter(
```
Comment (ptitloup) [inline pod/video/views.py]: Why not use VIDEOS ?

File: pod/video/views.py [hunk 8]
```diff
@@ -1282,3 +1286,111 @@ def video_oembed(request):
...
+                    encoding_in_progress=False,
+                    is_draft=False,
+                    channel=channel)
+    if target.lower() == "theme" and not videos and p_slug_t:
+        theme = Theme.objects.filter(
+                slug__istartswith=p_slug_t, channel__slug__istartswith=p_slug
+                ).first()
+        title = "Video viewing statics of '"+theme.title+"' theme"
+        if theme:
+            videos = Video.objects.filter(
```
Comment (ptitloup) [inline pod/video/views.py]: VIDEOS

File: pod/video/views.py [hunk 9]
```diff
@@ -1282,3 +1286,111 @@ def video_oembed(request):
...
+        videos.append(
+                Video.objects.filter(
+                    slug__istartswith=p_slug).first())
+        title = "'"+videos[0].title.capitalize()+"' video viewing statistics"
+    if target.lower() == "chaine" and not videos:
+        channel = Channel.objects.filter(
+                slug__istartswith=p_slug).first()
+        title = "Video viewing statistics of '"+channel.title+"' channel"
+        if channel:
+            videos = Video.objects.filter(
```
Comment (ptitloup) [inline pod/video/views.py]: global VIDEOS

File: pod/video/views.py [hunk 10]
```diff
@@ -1282,3 +1286,111 @@ def video_oembed(request):
...
+                slug__istartswith=p_slug_t, channel__slug__istartswith=p_slug
+                ).first()
+        title = "Video viewing statics of '"+theme.title+"' theme"
+        if theme:
+            videos = Video.objects.filter(
+                    encoding_in_progress=False,
+                    is_draft=False,
+                    theme=theme)
+    if not videos:
+        videos = [v for v in Video.objects.filter(
```
Comment (ptitloup) [inline pod/video/views.py]: VIDEOS

File: pod/video/views.py [hunk 11]
```diff
@@ -1282,3 +1286,111 @@ def video_oembed(request):
...
+        if not videos:
+            return HttpResponseNotFound(
+                    "La chaine ou la vidéo suivante n'existe pas : %s" % slug)
+        target = _(" of " + target)
+        return render(
+                request,
+                "videos/video_stats_view.html",
+                {"title": title})
+    specific_date = request.POST.get("periode", TODAY)
+    min_date = Video.objects.filter(
```
Comment (ptitloup) [inline pod/video/views.py]: VIDEOS

File: pod/video/templates/videos/video_stats_view.html (hunk 5 above)
Comment (meschac38700) [reply pod/video/templates/videos/video_stats_view.html]: Only Opera and QQ Browser don't support "let" => https://caniuse.com/#search=let

Comment (ptitloup) [reply pod/video/templates/videos/video_stats_view.html]: ok, thanks. Have you tried your JS code in several browser ?

**Comment by ptitloup** [comment]
Merci !

**Pull request merged**